Shared/Extensions/FontExtensions.swift

```diff
@@ -0,0 +1,21 @@
+// Copyright 2021 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import UIKit
+
+extension UIFont {
+    
+    /// Extended Dynamic Font Function for custom weight
+    /// - Parameters:
+    ///   - style: textStyle for the dynamic font
+    ///   - weight: weight of the dynamic font to be adjusted
+    public static func preferredFont(for style: TextStyle, weight: Weight) -> UIFont {
```

Comment:
I would move somewhere to to `BraveUI` framework,
`Shared` is a legacy framework inherited from FF

Reply:
Moved out of Shared

Fix at https://github.com/brave/brave-ios/pull/4124/commits/ded5065d24f539a7dee6bb8bcb40b93a2b70fe7b